test/integration/rmi_out_test.py

```diff
@@ -251,37 +183,40 @@ def test_consistency_of_data_stages(
         )
 
-    agg_test = agg_test_data(
+    agg_test = pudl_rmi.validate.agg_test_data(
         df1=rmi_out.__getattribute__(df1_name)(),
         df2=rmi_out.__getattribute__(df2_name)(),
-        data_col=data_col,
+        data_cols=data_cols,
         by=by,
     )
-    # there are a ton of unconnected data in these tables (eg. depreciation
-    # records that don't have a EIA connection) these records will not get
-    # effectively propegated through each of these stages that require a match
-    # to EIA. the `data_ratio` column being null is an indication that either df1
-    # or df2 has a null data_col.
-    actual_aggregation_errors = agg_test[
-        ~agg_test.data_isclose & agg_test.data_ratio.notnull()
-    ]
-
-    logger.info(
-        f"Failures for {data_col} by {by_name} btwn {df1_name} "
-        f"and {df2_name}: {len(actual_aggregation_errors)}"
-    )
-    expected_errors_path = (
-        EXPECTED_ERRORS_DIR
-        / f"expected_errors_{data_col}_{by_name}_{df1_name}_vs_{df2_name}.csv"
-    )
-    expected_aggregation_errors = pd.read_csv(expected_errors_path).set_index(by)
-    # the commented out lines here are here to help
-    # try:
-    pd.testing.assert_index_equal(
-        actual_aggregation_errors.index,
-        expected_aggregation_errors.index,
-        exact="equiv",
-        check_order=False,
-    )
-    # except AssertionError:
-    #     actual_aggregation_errors.reset_index()[by].to_csv(expected_error_path, index=False)
+    for data_col in data_cols:
+        # there are a ton of unconnected data in these tables (eg. depreciation
+        # records that don't have a EIA connection) these records will not get
+        # effectively propegated through each of these stages that require a match
+        # to EIA. the `data_ratio` column being null is an indication that either df1
+        # or df2 has a null data_col.
+        actual_aggregation_errors = agg_test[
+            ~agg_test[f"{data_col}_isclose"] & agg_test[f"{data_col}_ratio"].notnull()
+        ]
+
+        logger.info(
+            f"Failures for {data_col} by {by_name} btwn {df1_name} "
+            f"and {df2_name}: {len(actual_aggregation_errors)}"
+        )
+        expected_errors_path = (
+            EXPECTED_ERRORS_DIR
+            / f"expected_errors_{data_col}_{by_name}_{df1_name}_vs_{df2_name}.csv"
+        )
+        expected_aggregation_errors = pd.read_csv(expected_errors_path).set_index(by)
+        # the commented out lines here are here to help
+        # try:
+        pd.testing.assert_index_equal(
+            actual_aggregation_errors.index,
+            expected_aggregation_errors.index,
+            exact="equiv",
+            check_order=False,
+        )
+        # except AssertionError:
+        #     actual_aggregation_errors.reset_index()[by].to_csv(
+        #         expected_errors_path, index=False
+        #     )
```

Comment:
Are the errors typically going to be too voluminous to log? If this will typically get run in CI and you want to see what failed it might be useful to print it out. Or you can ask GitHub to save an "artifact" in the CI workflow -- set the CSV aside for later investigation.

Reply:
the pandas error message on `pd.testing.assert_index_equal` are not super huge. But it is still  a little challenging to pick out hte specific utilities or plants that changed.

Reply:
I left this in here because I VERY MUCH expect these known baddies to change whenever we change any of the connections between these three datasets. So we need a nimble enough way to update these bb files.